src/client.rs

```diff
@@ -153,14 +154,15 @@ impl<C: Context> Client<C> {
 
     /// Returns the metadata information for the specified topic, or for all topics in the cluster
     /// if no topic is specified.
-    pub fn fetch_metadata(&self, topic: Option<&str>, timeout_ms: i32) -> KafkaResult<Metadata> {
+    pub fn fetch_metadata(&self, topic: Option<&str>, timeout: Option<Duration>) -> KafkaResult<Metadata> {
         let mut metadata_ptr: *const RDKafkaMetadata = ptr::null_mut();
         let (flag, native_topic) = if let Some(topic_name) = topic {
             (0, Some(self.native_topic(topic_name)?))
         } else {
             (1, None)
         };
         trace!("Starting metadata fetch");
+        let timeout_ms = timeout_to_ms(timeout);
```

Comment:
nit: if `timeout_ms` is only used once, could we use the `timeout_to_ms` function directly instead? (same in the other methods)